apps/ledger-live-desktop/src/renderer/screens/nft/Gallery/Collection.jsx

```diff
@@ -81,7 +81,7 @@ const Collection = () => {
     if (isAtBottom && maxVisibleNTFs < nfts.length) {
       setMaxVisibleNFTs(maxVisibleNTFs => maxVisibleNTFs + 5);
     }
-  }, [isAtBottom]);
+  }, [isAtBottom, maxVisibleNTFs, nfts.length]);
```

Comment:
This looks wrong to me. If `useEffect` depends on `maxVisibleNTFs` and also calls `setMaxVisibleNFTs` it will change the original behaviour and loop..

This is typically a case which demonstrates that we should be careful with `react-hooks/exhaustive-deps` and not follow the rule blindly. It is more of a helper.

Reply:
@jelbaz-ledger Thanks, I really wasn't sure of the impact.

Do you think we should rather add an `// eslint-disable-next-line react-hooks/exhaustive-deps` on each of these cases? Or keep the rule as a warning?...

I thought having the rule as an error by default would be useful, as conditional hooks are an easy mistake to make, that can easily slip through review, and recently caused https://ledgerhq.atlassian.net/browse/LIVE-3599 for example.

Reply:
> I thought having the rule as an error by default would be useful, as conditional hooks are an easy mistake to make, that can easily slip through review, and recently caused https://ledgerhq.atlassian.net/browse/LIVE-3599 for example.

No worries I know the intention was good 🤗.

The thing is that the opposite is also true, enforcing the rule at all cost will cause bugs. There are of plenty of hard-to-detect cases where it really inefficient in terms of performance and also error prone to add all dependencies of a hook.

> Do you think we should rather add an // eslint-disable-next-line react-hooks/exhaustive-deps on each of these cases? Or keep the rule as a warning?...

With that said, I'm fine with setting `error` as the severity and adding an "ignore" comment, since the goal is to have 0 warnings / errors whatsoever when linting (and on the CI).